WcaOnRails/app/models/competition.rb

```diff
@@ -233,6 +234,7 @@ def update_foreign_keys
       CompetitionDelegate,
       CompetitionOrganizer,
       DelegateReport,
+      CompetitionTab,
     ].each do |model|
       foreign_key = model.column_names.include?("competitionId") ? "competitionId" : "competition_id"
       model.where(foreign_key => id_was).update_all(foreign_key => id)
```

Comment:
@larspetrus I remember you've suggested to automate this. What do you think about the following:
```ruby
  def update_foreign_keys
    Competition.reflect_on_all_associations.uniq(&:klass).each do |association_reflection|
      foreign_key = association_reflection.foreign_key
      if ["competition_id", "competitionId"].include?(foreign_key)
        association_reflection.klass.where(foreign_key => id_was).update_all(foreign_key => id)
      end
    end
  end
```

And the following test (passes):
```ruby
  it "when id is changed, foreign keys are updated as well" do
    competition = FactoryGirl.create(:competition, :with_delegate, :with_organizer, :with_delegate_report, :registration_open)
    FactoryGirl.create(:result, competitionId: competition.id)
    FactoryGirl.create(:competition_tab, competition: competition)
    FactoryGirl.create(:registration, competition: competition)

    expect do
      competition.update_attribute(:id, "NewName2016")
    end.to_not change {
      [:results, :organizers, :delegates, :competition_tabs, :registrations, :delegate_report].map do |associated|
        competition.send(associated)
      end
    }
  end
```

Reference:
- [reflect_on_all_associations](http://apidock.com/rails/v4.2.1/ActiveRecord/Reflection/ClassMethods/reflect_on_all_associations)
- [klass](http://apidock.com/rails/v4.2.1/ActiveRecord/Reflection/AssociationReflection/klass)


If you think it's to messy we can leave the current version but I worry that someone will forget about updating this method (fortunately you remembered me this time).